synapse/storage/databases/main/room.py

```diff
@@ -2389,7 +2389,7 @@ async def block_room(self, room_id: str, user_id: str) -> None:
             table="blocked_rooms",
             keyvalues={"room_id": room_id},
             values={},
-            insertion_values={"user_id": user_id},
+            insertion_values={"user_id": user_id or ""},
```

Comment:
Should we do that or drop the `NON NULL` constraint instead ?

Reply:
it's a pain to drop the NOT NULL constraint, especially with SQLite... but luckily the module doesn't even need to hit this path, so I'll just block it off instead!